build: 更新 Docker 构建配置和应用属性

YuanJie2001 opened the pull request:
nacos镜像优化
1.剔除原jdk8,使用eclipse adoptium-jdk11
2.使用ubuntu22.04取代alpone
3.使用G1取代cms垃圾回收器
其中jdk和nacos均为本地下载使用
jdk: https://adoptium.net/zh-CN/temurin/releases/?os=linux&version=11&arch=x64
 nacos：https://github.com/alibaba/nacos/releases/tag/2.3.2
![image](https://github.com/user-attachments/assets/e7e2219d-542e-46b3-982f-62e3c1254d5b)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/nacos-group/nacos-docker?pullRequest=450) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/nacos-group/nacos-docker?pullRequest=450) before we can accept your contribution.<br/><hr/>**YuanJie** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/nacos-group/nacos-docker?pullRequest=450) it.</sub>

YuanJie2001 commented:
对Docker基础镜像的思考，该不该选择alpine: https://cloud.tencent.com/developer/article/2168079

Commit:
build:重构 Dockerfile 并优化应用配置

- 重构 Dockerfile，分为构建阶段和运行时阶段，减小最终镜像大小
- 更新 application.properties，设置默认数据库连接参数和鉴权配置- 修改 docker-startup.sh，优化 JVM 参数配置
